constantinCurtis/src/main/java/data/collection/task/classes/AbstractUser.java

```diff
@@ -0,0 +1,70 @@
+package data.collection.task.classes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class AbstractUser {
+    private final String id;
+    private final String username;
+    private Map<String, Address> addresses;
+
+
+    public AbstractUser(String username, Map<String, Address> addresses) {
+        this.id = randomUUID();
+        this.username = username;
+        this.addresses = addresses;
+    }
+
+    public AbstractUser(String username) {
+        this.id = randomUUID();
+        this.username = username;
+    }
+
+    public String randomUUID() {
+        return String.valueOf(UUID.randomUUID());
+    }
+
+    @Override
+    public String toString() {
+        return "AbstractUser{" +
+                "id=" + id +
+                ", username='" + username + '\'' +
+                ", addresses=" + addresses +
+                '}';
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public Map<String, Address> getAddresses() {
+        return addresses;
+    }
+
+    public void setAddresses(Map<String, Address> addresses) {
+        this.addresses = addresses;
+    }
+
+    // create method to filter list of users whose city is Chisinau
+    public static List<User> filterUsersByCity(List<User> users, String city) {
```

Comment:
Can we use this method in order to filter a Company by its city? Could you please try it? If the answer is no, please refactor the method

Reply:
changed, thanks